mysys/crc32/crc32_arm64.c

```diff
@@ -18,8 +18,21 @@ int crc32_aarch64_available(void)
   unsigned long auxv= getauxval(AT_HWCAP);
   return (auxv & HWCAP_CRC32) != 0;
 }
+
+#if defined(HAVE_ARMV8_CRYPTO)
+
+#ifndef HWCAP_PMULL
+#define HWCAP_PMULL (1 << 4)
 #endif
 
+/* Check if target ARM machine support crc32 + pmull for computing crc32c */
+int crc32c_aarch64_available(void)
+{
+  return (crc32_aarch64_available() && (getauxval(AT_HWCAP) & HWCAP_PMULL));
```

Comment:
I think that it would be shorter to invoke `getauxval()` only once:
```c
int crc32c_aarch64_available(void)
{
  return !(~getauxval(AT_HWCAP) & (HWCAP_CRC32 | HWCAP_PMULL));
}
```
I would expect the generated code to be roughly the same size as `crc32_aarch64_available()` (maybe one instruction more to invert the bits before the bitwise-and operation). The code that you propose looks longer.

I tested with `gcc -O2` (using GCC 9.2.1) on https://godbolt.org and it seems to me that the generated code for your version of the run-time check would be almost twice the size of the one for my suggestion. This is with the assumption that `getauxval()` is a function call and not something that the compiler knows about. The headers are not available in that environment.

Reply:
right before submission, I was trying the same and it literally turned out to be 1 extra instruction (https://godbolt.org/z/fo1ccM) but may not translate to cycle saving but then I see we can re-use the said crc32 function which is more natural than replicating the code (but extra getauxval call).

But if you prefer I don't have reservation for proposed solution too. Anyway the function is invoked only once during boot and not on performance critical path.